pkg/venafi/tpp/connector.go

```diff
@@ -1189,15 +1206,28 @@ func (c *Connector) setContact(tppPolicy *policy.TppPolicy) (status string, err
 	return status, nil
 }
 
-func (c *Connector) resolveContacts(contacts []string) ([]string, error) {
-	var identities []string
-	uniqueContacts := getUniqueStringSlice(contacts)
+func (c *Connector) resolvePrefixedUniversals(filters []string) ([]string, error) {
+	var prefixedUniversals []string
+	identities, err := c.resolveIdentities(filters)
+	if err != nil {
+		return nil, err
+	}
+	for _, identityEntry := range identities {
+		prefixedUniversals = append(prefixedUniversals, identityEntry.PrefixedUniversal)
+	}
+
+	return prefixedUniversals, nil
+}
+
+func (c *Connector) resolveIdentities(filters []string) ([]*IdentityEntry, error) {
+	var identities []*IdentityEntry
+	uniqueContacts := getUniqueStringSlice(filters)
 	for _, contact := range uniqueContacts {
-		identity, err := c.getIdentity(contact)
+		identityEntry, err := c.getIdentity(contact)
```

Comment:
This can't work for emails since `getIdentities` checks that the queried filter is equal to the username. When the queried filter is an email, this won't work.

Reply:
please check the updated patch in my last comment, I think is solving this issue :-)

Reply:
Thanks for the patch. I now understand better why the test `identity.Name == filter` has been introduced a while back. I detailed this in https://github.com/Venafi/vcert/pull/221#issuecomment-1911747389.